### PR TITLE
CORTX-30463: Make use of static endpoint with static NodePorts

### PR DESCRIPTION
**Pull request by rahul27kumar**
Signed-off-by: Rahul Kumar <rahul.kumar@seagate.com>

## Problem Statement

To make a static endpoint, you must required one IP from host machine and port number. This IP should be define in /etc/hosts configuration file like below:

`xx.xx.xx.xx s3.seagate.com sts.seagate.com iam.seagate.com`
Note: Please ensure that IP shouldn't be management IP. (Basically ignore 1st ethernet IP that's management IP)

To define port, we can consider the following way to support:

- Read a `NodePort ` from `solution.yaml` and redirect the request based on `http ` or `https`.
- If user didn't define port as part of workload endpoint, then make use of default port of LC cluster.
- Provide flexibility to end user to use either `http://s3.seagate.com` or `https://s3.seagate.com`. 
- Previouly rgw addb logs was generating in `rgw_debug` location and now
  it is generating inside addb_files-*. Accordingly appiled fixes for it.
- Defined standard pre-defined workload as a part of nightly build jobs.

## Coding

-   [x] Coding conventions are followed and code is consistent
-   [x] No new Codacy Issues are added
-   [x] Version is updated in the respective VERSION file, format - `<tool>_v<major>.<minor>.<patch>`

## Documentation

-   [ ] Changes done to readme file / Quick Start Guide
-   [ ] Docstrings are added to new functions/ files with Summary, Args and Return params

## Testing

-   [x] New/Affected tests are executed on Latest Build
-   [x] Attach test execution logs
-   [x] No regression introduced

## Pull Request

-   [x] JIRA number/GitHub Issue added to PR, format - `<ID>: <Title>`
-   [x] Only one commit is present, format -
    ```txt
    <ID>: <Commit message>
    - bullets of changes committed in the PR
    ```
-   [x] Commit is sign-offed

## Review Checklist

-   [x] PR is self reviewed
-   [x] JIRA ticket state/status is updated
-   [x] Check if the DoD, description is clear and explained

## Impact Analysis

  (Checklist for Author/Reviewer/GateKeeper)

-   [ ] Are there any changes in any common function?

  If yes,

-   [ ] Relative/ dependent function calls are updated
-   [ ] Executed all affected tests

> Note: If checkboxes are not working, write Y/N in front of the bullets. Thanks for your contribution and Happy Coding!

### CODE REVIEW

File: performance/PerfLine/roles/perfline_setup/files/wrapper/scripts/s3bench_run.sh [hunk 1]
```diff
@@ -23,8 +23,9 @@ set -x
 SCRIPT_PATH="$(readlink -f "$0")"
 SCRIPT_DIR="${SCRIPT_PATH%/*}"
 source "$SCRIPT_DIR/../../perfline.conf"
-
-
+SOLUTION_CONFIG="$CORTX_K8S_REPO/k8_cortx_cloud/solution.yaml"
+HTTP_REQUEST=$(cat $SOLUTION_CONFIG | grep -m 1 -A2 "nodePorts" | grep -w "http" | awk '{ print $2 }' )
```
Comment (rkothiya) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/scripts/s3bench_run.sh:27]: To get rid of codacy warning, please put $SOLUTION_CONFIG in double quotes.

File: performance/PerfLine/roles/perfline_setup/files/wrapper/scripts/s3bench_run.sh [hunk 2]
```diff
@@ -23,8 +23,9 @@ set -x
...
+HTTPS_REQUEST=$(cat $SOLUTION_CONFIG | grep -m 1 -A2 "nodePorts" | grep -w "https" | awk '{ print $2 }' )
```
Comment (rkothiya) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/scripts/s3bench_run.sh:28]: To get rid of codacy warning, please put $SOLUTION_CONFIG in double quotes.

**Review by rkothiya — COMMENTED** [review]
I tried a snip from the code on my cluster it is getting the ip and the port. 
However it would be really nice if we paste the output of the test result in the ticket.

**Comment by rahul27kumar** [comment]
Getting S3_ENDPOINT_IP:
```
[root@ssc-vm-rhev4-2700 PerfLine]# S3_ENDPOINT_IP=$(ssh "ssc-vm-rhev4-2700" ip -f inet -br addr | grep -m 2 'UP' | awk '{ print $3 }' | cut -d '/' -f1 | tail -1)
[root@ssc-vm-rhev4-2700 PerfLine]# echo $S3_ENDPOINT_IP
192.168.51.118
[root@ssc-vm-rhev4-2700 PerfLine]#
```

Extract a nodeports from solution.yaml:
```
[root@ssc-vm-rhev4-2700 PerfLine]# SOLUTION_CONFIG="/root/deploy-scripts/k8_cortx_cloud/solution.yaml"
[root@ssc-vm-rhev4-2700 PerfLine]# HTTP_REQUEST=$(cat "$SOLUTION_CONFIG" | grep -m 1 -A2 "nodePorts" | grep -w "http" | awk '{ print $2 }' )
[root@ssc-vm-rhev4-2700 PerfLine]# HTTPS_REQUEST=$(cat "$SOLUTION_CONFIG" | grep -m 1 -A2 "nodePorts" | grep -w "https" | awk '{ print $2 }' )
[root@ssc-vm-rhev4-2700 PerfLine]# echo $HTTP_REQUEST
30080
[root@ssc-vm-rhev4-2700 PerfLine]# echo $HTTPS_REQUEST
30443
[root@ssc-vm-rhev4-2700 PerfLine]#
```

**Review by rkothiya — APPROVED** [review]
LGTM

**Review by rajeshdesh — APPROVED** [review]

**Pull request merged**